homeassistant/loader.py

```diff
@@ -290,13 +290,37 @@ def resolve_from_root(
                 )
                 continue
 
-            return cls(
+            integration = cls(
                 hass,
                 f"{root_module.__name__}.{domain}",
                 manifest_path.parent,
                 manifest,
             )
 
+            if not integration.is_built_in:
```

Comment:
consider converting this to a guard clause because the rest of this code branch is just this check

```python
if integration.is_built_in:
    return integration
```